src/functions/TestDrive.ps1

```diff
@@ -24,39 +43,76 @@
                     TestDrivePath    = $testDrivePath
                 })
         }
-    } -EachBlockTearDownEnd {
+    }
+
+    $p.EachBlockTearDownEnd = {
         param($Context)
 
+        # Remap drive and variable if missing/wrong? Ex. if nested run was cancelled and didn't re-attach this drive
+        if (-not (& $script:SafeCommands['Test-Path'] TestDrive:\)) {
+            New-TestDrive -Path $Context.Block.PluginData.TestDrive.TestDrivePath
+        }
+
         if ($Context.Block.IsRoot) {
             # this is top-level block remove test drive
             Remove-TestDrive -TestDrivePath $Context.Block.PluginData.TestDrive.TestDrivePath
         }
         else {
-            Clear-TestDrive -TestDrivePath $Context.Block.PluginData.TestDrive.TestDrivePath -Exclude ( $Context.Block.PluginData.TestDrive.TestDriveContent )
+            Clear-TestDrive -TestDrivePath $Context.Block.PluginData.TestDrive.TestDrivePath -Exclude ($Context.Block.PluginData.TestDrive.TestDriveContent)
         }
     }
+
+    $p.End = {
+        param($Context)
+
+        if ($Context.GlobalPluginData.TestDrive.ExistingTestDrivePath) {
+            # If nested run, reattach previous TestDrive PSDrive and variable
+            New-TestDrive -Path $Context.GlobalPluginData.TestDrive.ExistingTestDrivePath
+        }
+    }
+
+    New-PluginObject @p
 }
 
-function New-TestDrive ([Switch]$PassThru, [string] $Path) {
```

Comment:
PSScriptAnalyzer doesn't track parameters properly without a param-block it seems, so changed a few functions. Ex. this has a unused `$PassThru` switch. Removed since this is internal.